### ISSUE TITLE
Proposal: Remove `and` from the syntax

### ISSUE THREAD

**Issue by Marwes** [gluon-lang/gluon#322]
Currently then `and` keyword is used to define additional mutually recursive bindings. While this is simple to understand and easy to parser I found that it complicates things a bit when grepping code. Often I want to see where something is defined but as it is possible to define a binding with both `let`/`type` as well as `and` it is not possible to do `grep "let binding"` to find a binding for certain similarily of doing `grep fn binding` in rust.

An idea I had is to instead assume that all function `let` bindings that follow one another are mutually recursive.

```f#
let f x = g x
and g x = f x
// vs
let f x = g x
let g x = f x
```

This solves the "grep problem" but I worry that it makes it harder to reason about when bindings are mutually recursive.

```f#
// Mutually recursive
let f x = g x
let g x = f x

// Error undefined `g`
let f x = g x
let value = 1
let g x = f x
```

Input welcome!

**Comment by brendanzab** [comment]
Why not just make the entire module mutually recursive? :trollface: 

Or do what Idris does and mark it:

```
mutual
    let f x = g x
    let g x = f x
```

**Comment by Marwes** [comment]
I can see the trollface :), but that can only actually work for pure + lazy languages which complicates things a bit.

`mutual` may be less confusing but the block indent feels out of place for something as simple as recursive functions...

**Comment by Marwes** [comment]
If we were to add lazy bindings then having `mutual` would make that unambiguous as well.

```f#
mutual
    // Create a lazy list of an infinite amount of `1`
    let ones = LazyCons 1 ones
```

**Comment by Marwes** [comment]
Closed by #614
